transaction.py

```diff
@@ -257,27 +257,43 @@ def tr_error(self, tr):
         self._running_flushes -= 1
         self._finished_flushes += 1
         tr.inc_error_count()
-        tr.compute_next_flush(self._MAX_WAIT_FOR_REPLAY)
-        log.warn("Transaction %d in error (%s error%s), it will be replayed after %s",
-                 tr.get_id(),
-                 tr.get_error_count(),
-                 plural(tr.get_error_count()),
-                 tr.get_next_flush())
-        self._endpoints_errors[tr._endpoint] = self._endpoints_errors.get(tr._endpoint, 0) + 1
-        # Endpoint failed too many times, it's probably an enpoint issue
-        # Let's avoid blocking on it
-        if self._endpoints_errors[tr._endpoint] == self._MAX_ENDPOINT_ERRORS:
-            new_trs_to_flush = []
-            for transaction in self._trs_to_flush:
-                if transaction._endpoint != tr._endpoint:
-                    new_trs_to_flush.append(transaction)
-                else:
-                    transaction.compute_next_flush(self._MAX_WAIT_FOR_REPLAY)
-            log.debug('Endpoint %s seems down, removed %s transaction from current flush',
-                      tr._endpoint,
-                      len(self._trs_to_flush) - len(new_trs_to_flush))
-
-            self._trs_to_flush = new_trs_to_flush
+        if tr.get_error_count() > self._MAX_ENDPOINT_ERRORS:
+            log.warn("Transaction %d failed too many (%d) times, removing",
+                     tr.get_id(),
+                     tr.get_error_count())
+            self._remove(tr)
+            self._transactions_flushed += 1
+            self.print_queue_stats()
```

Comment:
I imagine that this would be something useful to see, this is on purpose, correct?

Reply:
yes and no, meaning that i want this error to do the same as tr_rejection_error, which included the print_queue_stats(). I think it is useful